pkg/types/powervs/machinepools.go

```diff
@@ -0,0 +1,65 @@
+package powervs
+
+// ProcType defines valid types for a ppc64le processor in Power VS
+// +kubebuilder:validation:Enum="",capped;dedicated;shared
+type ProcType string
+
+// Capped type for capped processor consumption
+const Capped ProcType = "capped"
+
+// Dedicated type for dedicated processor(s)
+const Dedicated ProcType = "dedicated"
+
+// Shared type shared type for shared processor(s)
+const Shared ProcType = "shared"
```

Comment:
Tie these together in a `const` block so that it is easier to see that they are all of the values of `ProcType`.
```suggestion
const (
	// Capped type for capped processor consumption
	Capped ProcType = "capped"

	// Dedicated type for dedicated processor(s)
	Dedicated ProcType = "dedicated"

	// Shared type shared type for shared processor(s)
	Shared ProcType = "shared"
)
```